File: translationRecorder/app/src/main/java/org/wycliffeassociates/translationrecorder/SettingsPage/UploadServerDialog.java
package org.wycliffeassociates.translationrecorder.SettingsPage;

import android.app.Dialog;
import android.app.DialogFragment;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import androidx.appcompat.app.AlertDialog;
import org.wycliffeassociates.translationrecorder.R;
import org.wycliffeassociates.translationrecorder.TranslationRecorderApp;
import org.wycliffeassociates.translationrecorder.database.ProjectDatabaseHelper;

/**
 * Created by sarabiaj on 12/14/2016.
 */

public class UploadServerDialog extends DialogFragment {

    SharedPreferences pref;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        View view = getActivity().getLayoutInflater().inflate(R.layout.dialog_upload_server, null);

        pref = PreferenceManager.getDefaultSharedPreferences(getActivity());
        String currentServerName = pref.getString(Settings.KEY_PREF_UPLOAD_SERVER, "opentranslationtools.org");

        final EditText serverName = (EditText) view.findViewById(R.id.server_name);
        final Button saveButton = (Button) view.findViewById(R.id.save_button);
        final Button restoreButton = (Button) view.findViewById(R.id.restore_default);
        final Button cancelButton = (Button) view.findViewById(R.id.close_button);

        serverName.setText(currentServerName);

        saveButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String name = serverName.getText().toString();

                if(name.length() > 0) {
                    pref.edit().putString(Settings.KEY_PREF_UPLOAD_SERVER, name).commit();
                    dismiss();
                }
            }
        });

        restoreButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                pref.edit().putString(Settings.KEY_PREF_UPLOAD_SERVER, "opentranslationtools.org").commit();
                dismiss();
            }
        });

        cancelButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dismiss();
            }
        });

        builder.setView(view);
        return builder.create();
    }
}

File: translationRecorder/app/src/main/java/org/wycliffeassociates/translationrecorder/SettingsPage/UploadServerDialog.kt
package org.wycliffeassociates.translationrecorder.SettingsPage

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import dagger.hilt.android.AndroidEntryPoint
import org.wycliffeassociates.translationrecorder.R
import org.wycliffeassociates.translationrecorder.databinding.DialogUploadServerBinding
import org.wycliffeassociates.translationrecorder.persistance.IPreferenceRepository
import org.wycliffeassociates.translationrecorder.persistance.getDefaultPref
import org.wycliffeassociates.translationrecorder.persistance.setDefaultPref
import javax.inject.Inject

/**
 * Created by sarabiaj on 12/14/2016.
 */
@AndroidEntryPoint
class UploadServerDialog : DialogFragment() {
    @Inject lateinit var pref: IPreferenceRepository

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(requireActivity())
        val binding = DialogUploadServerBinding.inflate(layoutInflater)

        val currentServerName = pref.getDefaultPref(
            SettingsActivity.KEY_PREF_UPLOAD_SERVER,
            getString(R.string.pref_upload_server)
        )

        binding.serverName.setText(currentServerName)

        binding.saveButton.setOnClickListener {
            val name = binding.serverName.text.toString()
            if (name.isNotEmpty()) {
                pref.setDefaultPref(SettingsActivity.KEY_PREF_UPLOAD_SERVER, name)
                dismiss()
            }
        }

        binding.restoreDefault.setOnClickListener {
            pref.setDefaultPref(
                SettingsActivity.KEY_PREF_UPLOAD_SERVER,
                getString(R.string.pref_upload_server)
            )
            dismiss()
        }

        binding.closeButton.setOnClickListener { dismiss() }

        builder.setView(binding.root)
        return builder.create()
    }
}
